docs/lib/introduction.md

```diff
@@ -0,0 +1,789 @@
+[_Credential Repository Types_](../../pkg/contexts/credentials/repositories) that have to implement the following
+[interface](../../pkg/contexts/credentials/internal/repository.go):
+
+```
+type Repository interface {
+   ExistsCredentials(name string) (bool, error)
+   LookupCredentials(name string) (Credentials, error)
+   WriteCredentials(name string, creds Credentials) (Credentials, error)
+}
+```
+
+### Section 3.2.2 - Access Type Scheme
+
+An `AccessTypeScheme` is an object that maps a number of _access types (= model type)_ to corresponding
+_factories_ providing an access spec object which is again a _factory_ for providing an object of a specific type or
+with
+a specific configuration for dealing with this _access type (= model type)_.
+
+From the perspective of the ocm-lib, any _go type_ that implements the following
+[interface](../../pkg/contexts/ocm/internal/accesstypes.go) can be considered an _OCI Repository_:
+
+```
+type AccessMethod interface {
+   DataAccess
+
+   GetKind() string
+   AccessSpec() AccessSpec
+   MimeType
+   Close() error
+}
+```
+
+Within ocm-lib, there are multiple implementations of this interface. Most prominently, an implementation for
+[_OCI Artifacts_](../../pkg/contexts/ocm/accessmethods/ociartifact/method.go) (artifact blob is stored as an independent
+oci artifact in an OCI Registry), a dedicated technical procedure of how to access an artifact blob stored in an OCI
+Registry. There are further implementations, e.g. for [_Local Blobs_](../../pkg/contexts/ocm/accessmethods/localblob)
+(artifact blob is stored in the same oci artifact as its component descriptor), for  
+[_s3_](../../pkg/contexts/ocm/accessmethods/s3) or for [_gitHub_](../../pkg/contexts/ocm/accessmethods/github).
+
+**CAREFUL - There are two conceptually relevant details here!**
+
+1) There are **Global** and **Local** **Access Method types**. **Global types** (such as ociArtifact or github)
+   implement the procedure to access the artifact blobs themselves. **Local types**, on the contrary, delegate the
+   implementation of the procedure to the respective component repository (as each type of component repository may
+   handle the storage of artifact blobs alongside the component descriptor differently).
+2) **AccessMethods return the artifact blob as is**. Thus, if a helm chart is stored as an .tar, the AccessMethod will
+   return the bytes of the .tar. If, on the other hand, the helm chart is stored as an tar.gz , the AccessMethod will
+   return the bytes of the .tar.gz. Therefore, for further processing, the MimeType() can be read (inspecting the
+   MimeType and conversion to a target format can be done with downloaders - a convenience feature implemented by the
+   ocm library kind of as an add-on).
+
+### Section 3.3 - Constraint Registries aka. Handler Registries (and Handler-Registration Handler Registries)
+
+**Constraint Registries aka. Handler Registries**  
```

Comment:
Why are these called _CONSTRAINT_ Registries ? That term seems a bit unintuitive, no ?